### PR TITLE
port from spells-mainnet

### PR DESCRIPTION
**Pull request by julienmartinlevrai**
The best way to review these changes is to diff them to current spell-mainnet files on master.

### CODE REVIEW

File: src/test/config.sol [hunk 1]
```diff
@@ -99,7 +99,7 @@ contract Config {
         //
         spellValues = SpellValues({
             deployed_spell:                 address(0xB273Cc14acAF229dAB3eD6AEf83A8bb352919D66),        // populate with deployed spell if deployed
```
Comment (brianmcmichael) [inline src/test/config.sol:101]:
```suggestion
            deployed_spell:                 address(0),        // populate with deployed spell if deployed
```

File: src/test/config.sol [hunk 2]
```diff
@@ -99,7 +99,7 @@ contract Config {
...
-            deployed_spell_created:         1664905308,                 // use tx=<deploy tx> make deploy-stamp
+            deployed_spell_created:         1664905308,                 // use `./scripts/get-created-timestamp.sh <deployment-tx>`
```
Comment (brianmcmichael) [inline src/test/config.sol:102]: Does the `make` command not work anymore?

File: src/test/config.sol [hunk 3]
```diff
@@ -628,7 +628,7 @@ contract Config {
             line:         0,
             dust:         10 * THOUSAND,
             pct:          400,
-            mat:          2430000,  // 24,300%
```
Comment (brianmcmichael) [inline src/test/config.sol:631]: Not sure you need to rm this comment

Comment (brianmcmichael) [reply src/test/config.sol:631]: It's such a weird number that it the comment helps to indicate that it's intentional

File: src/test/config.sol (hunk 2 above)
Comment (brianmcmichael) [reply src/test/config.sol:102]: nvm the script path is probably better here

File: src/test/config.sol (hunk 1 above)
Comment (julienmartinlevrai) [reply src/test/config.sol:101]: I did this in the spell PR

File: src/test/config.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (julienmartinlevrai) [reply src/test/config.sol:102]: I did this in the spell PR

File: src/test/config.sol (hunk 3 above)
Comment (julienmartinlevrai) [reply src/test/config.sol]: done

**Review by iamchrissmith — APPROVED** [review]
looks good to merge

**Review by The-Arbiter — COMMENTED** [review]
Looks good to me as it is now. Any reason why `testCollateralIntegrations` isn't public by default if you need to make it public to use it? I'm assuming this is for gas savings or that there is some reason for this.

**Review by brianmcmichael — APPROVED** [review]
Good to merge

**Pull request merged**